antsibull/build_ansible_commands.py

```diff
@@ -272,6 +276,35 @@ def build_single_command() -> int:
     build_file = BuildFile(app_ctx.extra['build_file'])
     build_ansible_version, ansible_base_version, deps = build_file.parse()
     ansible_base_version = PypiVer(ansible_base_version)
+
+    # If we're building a feature frozen release (betas and rcs) then we need to
+    # change the upper version limit to not include new features.
+    if app_ctx.extra['feature_frozen']:
+        old_deps, deps = deps, {}
+        # For each collection that's listed...
+        for collection_name, spec in old_deps.items():
+            spec = SemVerSpec(spec)
+            new_clauses = []
+            min_version = None
+
+            # Look at each clause of the version specification
+            for clause in spec.clause.clauses:
+                if clause.operator in ('<', '<='):
+                    # Omit the upper bound as we're replacing it
+                    continue
+
+                if clause.operator == '>=':
+                    # Save the lower bound so we can write out a new compatible version
+                    min_version = clause.target
+
+                new_clauses.append(str(clause))
+
+            if min_version is None:
+                raise ValueError(f'No minimum version specified for {collection_name}: {spec}')
+
+            new_clauses.append(f'<{min_version.major}.{min_version.minor + 1}.0')
+            deps[collection_name] = ','.join(new_clauses)
+
     included_versions = asyncio.run(get_collection_versions(deps, app_ctx.galaxy_url))
```

Comment:
I guess we need to define eventually when to pass `pre=True` here.

Reply:
Yep.  I think it's mostly a project policy question (are there times we want to allow pre-releases into any of our ansible releases?  rc's of collections, perhaps?) but there's some releng-y questions too... Should we allow pre-releases when building alphas and betas?  Should those scenarios just be manual changes?

I'm not sure of the answers so I coded this as a switch that can be toggled but also hardcoded the toggle for now since we don't need it at the moment.

Reply:
True. I fully agree with status quo :)